cli/binary/switch.go

```diff
@@ -0,0 +1,147 @@
+		Items:        supportedPrograms,
+		HideSelected: true,
+	}
+	_, program, err := programSelect.Run()
+
+	return program, err
+}
+
+// ChooseVersion shows a menu in terminal to choose version of program to switch to.
+func ChooseVersion(cliOpts *config.CliOpts, programName string) (string, error) {
+	binDirFilesList, err := os.ReadDir(cliOpts.Env.BinDir)
+
+	if len(binDirFilesList) == 0 || errors.Is(err, fs.ErrNotExist) {
+		return "", fmt.Errorf("there are no binaries installed in this environment of 'tt'")
+	} else if err != nil {
+		return "", fmt.Errorf("error reading directory %q: %s", cliOpts.Env.BinDir, err)
+	}
+	versions, err := ParseBinaries(binDirFilesList, programName, cliOpts.Env.BinDir)
+	if err != nil {
+		return "", err
+	}
+	if len(versions) == 0 {
+		return "", fmt.Errorf("there are no %s installed in this environment of 'tt'", programName)
+	}
+	var versionStr []string
+	for _, version := range versions {
+		if strings.Contains(version.Str, "[active]") {
+			versionStr = append(versionStr, util.Bold(color.GreenString(version.Str)))
+			continue
+		}
+		versionStr = append(versionStr, color.YellowString(version.Str))
+	}
+	versionSelect := promptui.Select{
+		Label:        "Select version",
+		Items:        versionStr,
+		HideSelected: true,
+	}
+	_, version, err := versionSelect.Run()
+	version = cleanString(version)
+	if strings.HasSuffix(version, " [active]") {
+		version = strings.TrimSuffix(version, " [active]")
+	}
+
+	return version, err
+}
+
+// switchTt switches 'tt' program.
+func switchTt(binDir string, switchCtx SwitchCtx) error {
+	log.Infof("Switching to %s %s.", switchCtx.ProgramName, switchCtx.Version)
+	versionStr := search.ProgramTt + version.FsSeparator + switchCtx.Version
+	if util.IsRegularFile(filepath.Join(binDir, versionStr)) {
+		err := util.CreateSymlink(versionStr, filepath.Join(binDir, "tt"), true)
+		if err != nil {
+			return fmt.Errorf("failed to switch version: %s", err)
+		}
+		log.Infof("Done")
+		return err
+	}
+	return nil
+}
+
+// switchTarantool switches 'tarantool' program.
+func switchTarantool(binDir string, includeDir string, switchCtx SwitchCtx, enterprise bool) error {
+	log.Infof("Switching to %s %s.", switchCtx.ProgramName, switchCtx.Version)
+	var versionStr string
+	if enterprise {
+		versionStr = search.ProgramEe + version.FsSeparator + switchCtx.Version
+	} else {
+		versionStr = search.ProgramCe + version.FsSeparator + switchCtx.Version
+	}
+	if util.IsRegularFile(filepath.Join(binDir, versionStr)) &&
+		util.IsDir(filepath.Join(includeDir, "include", versionStr)) {
+		err := util.CreateSymlink(versionStr, filepath.Join(binDir, "tarantool"), true)
+		if err != nil {
+			return err
+		}
+		err = util.CreateSymlink(versionStr, filepath.Join(includeDir,
+			"include", "tarantool"), true)
+		if err != nil {
+			return err
+		}
+		log.Infof("Done")
+		return err
+	}
+	return nil
+}
+
+// Switch switches binaries.
+func Switch(binDir string, includeDir string, switchCtx SwitchCtx,
```

Comment:
Looks like, this functionality is not covered by unit-tests.

Reply:
added